### PR TITLE
read bearer token on every request + some http and scrape tests

### PR DESCRIPTION
**Pull request by krasi-georgiev**
read bearer token on every request
removed unuseful  scrape manager startup log
new tests -TestScrapeManagerReloadNoChange( scrape pool is not reloaded
when the config hasn't changed), TestMissingBearerAuthFile ,
TestBearerAuthFileRoundTripper

### CODE REVIEW

**Comment by krasi-georgiev** [comment]
@Gouthamve it is ready. 
most of this is your code anyway :)

File: util/httputil/client_test.go [hunk 1]
```diff
@@ -272,6 +302,32 @@ func TestBearerAuthRoundTripper(t *testing.T) {
...
+
+	fakeRoundTripper := testutil.NewRoundTripCheckRequest(func(req *http.Request) {
+		bearer := req.Header.Get("Authorization")
+		if bearer != ExpectedBearer {
+			t.Errorf("The expected Bearer Authorization (%s) differs from the obtained Bearer Authorization (%s)",
+				ExpectedBearer, bearer)
+		}
+	}, nil, nil)
+
+	//Normal flow
```
Comment (Gouthamve) [inline util/httputil/client_test.go]: space after `//` and should end with `.`.

Same below.

Comment (krasi-georgiev) [reply util/httputil/client_test.go]: fixed. 
btw these were some comment from a someone elses commit, so no red points for me :)

Comment (krasi-georgiev) [reply util/httputil/client_test.go]: @Gouthamve ^

File: retrieval/manager.go [hunk 2]
```diff
@@ -56,8 +56,6 @@ type ScrapeManager struct {
 
 // Run starts background processing to handle target updates and reload the scraping loops.
 func (m *ScrapeManager) Run(tsets <-chan map[string][]*targetgroup.Group) error {
-	level.Info(m.logger).Log("msg", "Starting scrape manager...")
```
Comment (brancz) [inline retrieval/manager.go]: why was this removed?

File: util/httputil/client_test.go (hunk 1 above)
Comment (brancz) [reply util/httputil/client_test.go]: @krasi-georgiev it was only fixed in the comments below, not these

File: retrieval/manager.go (hunk 2 above)
Comment (krasi-georgiev) [reply retrieval/manager.go]: it is not useful at all and no other Managers has this. 

It is useful only for the tsdb  - starting - started
